packages/react-router-website/api/Prompt.md

```diff
@@ -34,3 +34,24 @@ Instead of conditionally rendering a `<Prompt>` behind a guard, you can always r
 ```js
 <Prompt when={formIsHalfFilledOut} message="Are you sure?"/>
 ```
+
+## beforeUnload: bool _NavigationPrompt_
+Prevent user for navigating away from the site. Prompt user to warn 
+about leaving the site.
+
+```js
+<NavigationPrompt beforeUnload={false} />
+```
+
+## beforeUnload: func _NavigationPrompt_
+Prevent user for navigating away from the site. Perform custom actions, 
+for example clean up code before prompting the user.
+
+```js
+<NavigationPrompt beforeUnload={() { 
+  console.log("custom action - i.e. saving data before leaving.")
+  var dialogText = "Changes you made may not be saved."
+  e.returnValue = dialogText
```

Comment:
If you're talking about using the return value, or providing a custom message for the dialog, then this is not possible in the latest browsers. This feature has been dropped. The message here is simply to support older browsers to make them consistent. The return value is overridden in modern browsers.

Starting with Firefox 4, Chrome 51, Opera 38 and Safari 9.1, a generic string not under the control of the webpage will be shown instead of the returned string.

https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload

Let me know if I have miss understood your comment.